Fix main

amphioxis opened the pull request:
- [x]  The file [pypiResolver](https://github.com/fasten-project/fasten-pypi-plugin/blob/main/src/fasten/pypiResolver.py) looks like an independent tool. Can that not just be imported?
- [ ]  How can the received licenses for the packages unknown by fasten be accessed?
- [ ]  Do I need to integrate them into the OSCG or will that analysis be done beforehand?
- [ ]  If so, we would need to analyze the OSCG for licenses again for the known packages?

I will produce the output related to license violations.
Thanks.

(Put that list back as otherwise the commentaries below doesn't make sense anymore)

michelescarlato commented:
pypiResolver is an independent tool, now imported.

Think about vulnerabilities, please. Licensing is not easy as you think.

amphioxis commented:
Will licenses (for packages unknown by FASTEN) be integrated into the `Optimized Stitched Call Graph`?

michelescarlato commented:
> Will licenses (for packages unknown by FASTEN) be integrated into the `Optimized Stitched Call Graph`?

Nope.

I will follow Thomas approach, to use a conceptual approach.

michelescarlato commented:
> Pls don't merge!
> 
> * [x]  The file [pypiResolver](https://github.com/fasten-project/fasten-pypi-plugin/blob/main/src/fasten/pypiResolver.py) looks like an independent tool. Can that not just be imported?
> * [ ]  How can the received licenses for the packages unknown by fasten be accessed?

For unknown packages I produce a dictionary with license information, with this [function](https://github.com/fasten-project/fasten-pypi-plugin/blob/f3459c7e782b15f886caa05dcecb32aa0109b5b9/src/fasten/retrieveLocallyLicensesInformation.py
) .

> * [ ]  Do I need to integrate them into the OSCG or will that analysis be done beforehand?
> * [ ]  If so, we would need to analyze the OSCG for licenses again for the known packages?

I will produce the output related to license violations.
Thanks.

lisa-noeth commented on src/fasten/__main__.py:
```diff
@@ -11,7 +11,7 @@
 from fasten.stitchedCallGraphAnalyzer import StitchedCallGraphAnalyzer
 from fasten.createDirectories import CreateDirectories
-from executePypiResolver import ExecutePypiResolver
+from fasten.executePypiResolver import ExecutePypiResolver
```
blabla

amphioxis commented:
Misleading topic. Sry!

amphioxis commented:
@michelescarlato 

You can't see my comments here, as I am the reviewer of my own pull request. I created a new Issue regarding the most important comment here.
Feel free to merge this Pull Request if the changes I made are ok.

michelescarlato commented:
@lisa-noeth , @amphioxis am I wrong, or do you want me to review a PR with only this modification:
https://github.com/fasten-project/fasten-pypi-plugin/pull/22/files

amphioxis commented:
Hey @michelescarlato 

this is a little bit messed up here. I also had a discussion with Lisa about this Pull Request as it is also misleading, so lets get rid of it as fast as possible.

There must be 3 commits - I removed trailing whitespaces, added `fasten` to the imported module and removed an additional call of the `depthFirstSearch` module.
I think you can just merge this. The review for you is just important that you now, what are the changes here and also apply them to your branches (especially the removed additional call of `depthFirstSearch`).
Sry for the confusion,

michelescarlato reviewed (approved):
I don't know how much sense have a PR for a small modification.
Anyway, let's try this approach.
